### ISSUE TITLE
Deprecate StreamingDataset and MutableDataset

### ISSUE THREAD

**Issue by jamesrkg** [Crunch-io/scrunch#372]
These classes were implemented before Crunch had a dedicated attribute for streaming. Now that it does the distinction of these streaming states as dedicated classes is not needed. Instead, protected methods can reference the streaming flag directly.

This should be done prior to working on https://github.com/Crunch-io/scrunch/issues/370.

**Comment by mathiasbc** [comment]
This is doable in the short term.

**Comment by xbito** [comment]
We do want to provide some sort of compatibility layer. Probably concentrate methods in the Dataset class and continue having Streaming and Mutable as subclasses of it, but raise a DeprecationWarning when making an instance out of them.

**Comment by mathiasbc** [comment]
Yes I was gonna propose a very similar thing. I should have a PR ready soon and we can discuss based off that.

**Comment by jamesrkg** [comment]
Sounds good to me.

**Comment by mathiasbc** [comment]
Initial PR: https://github.com/Crunch-io/scrunch/pull/376
